src/plotioda/plots/plots.py

```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+import numpy as np
+from emcpy.plots.plots import Scatter
+from emcpy.plots.map_plots import MapScatter
+from emcpy.plots import CreateMap, CreatePlot, VariableSpecs
+from emcpy.plots.map_tools import Domain, MapProjection
+
+__all__ = ['gen_figure']
+
+
+def gen_figure(plot, obsspace):
+    """
+    factory to generate figure depending on type of plot
+
+    Args:
+        plot: dictionary containing configuration
+        obsspace: IODA Python ObsSpace object
+
+    Returns:
+        fig: matplotlib figure object
+    """
+    plot_types = {
+        'map_scatter': _map_scatter,
+    }
+
+    fig = plot_types[plot['type']](plot, obsspace)
+
+    return fig
+
+
+def _map_scatter(plot, obsspace):
+    # plot a variable on a map
+    # get map dataframe
+    df = _gen_map_df(plot, obsspace)
+    print(df)
+
+    # define map domain/projection based off of YAML
+    domain = Domain(plot.get('domain', 'global'))
+    proj = MapProjection(plot.get('projection', 'plcarr'))
+
+    # generate map and add features
+    mymap = CreateMap(figsize=(12, 8), domain=domain, proj_obj=proj)
+    mymap.add_features(plot.get('features', ['coastlines']))
+
+    # generate data to plot and set plotting attributes
+    plotobj = MapScatter(df['latitude'], df['longitude'], df['variable'])
+    for attr in ['cmap', 'vmin', 'vmax']:
+        if attr in plot:
+            setattr(plotobj, attr, plot[attr])
+
+    # draw data on map
+    mymap.draw_data([plotobj])
+
+    # add titles, color bar, etc.
+    mymap.add_colorbar(label=plot.get('colorbar label', ''),
+                       label_fontsize=12, extend='neither')
+    mymap.add_title(label=plot.get('title', ''),
+                    loc='left', fontsize=12)
+    mymap.add_title(label=plot.get('cycle string', ''),
+                    loc='right', fontsize=12,
+                    fontweight='semibold')
+    mymap.add_xlabel(xlabel='Longitude')
+    mymap.add_ylabel(ylabel='Latitude')
+
+    # add a lat/lon grid if requested
+    if 'grid' in plot and plot['grid']:
```

Comment:
So would grid just be added to the yaml as an input if the user wanted it?

Reply:
Also, I found that this style of grid is way less intrusive on the plot if you would want to test and see:

`mymap.add_grid(linewidth=0.5, color='grey', linestyle='--')`

Reply:
Yes, the YAML would say `grid: true`, let me add that suggested color for the grid and push a new commit.